lib/dfe_sign_in/api/response.rb

```diff
@@ -0,0 +1,35 @@
+module DfeSignIn
+  module API
+    class Response
+      class NilUsersError < StandardError; end
+
+      attr_reader :response
+
+      def initialize(request)
+        @response = request.perform
+      end
+
+      def number_of_pages
+        raise(error_message) if response["numberOfPages"].nil?
+
+        response["numberOfPages"]
+      end
+
+      def users
+        raise(DfeSignIn::API::Response::NilUsersError, error_message) if users_nil_or_empty?
```

Comment:
As this error was always raised when iterating over the response users on a higher abstraction level, I have moved it directly to the `#users` call to improve cohesion and reduce the coupling from the class calling for the results.